extmod/modframebuf.c

```diff
@@ -491,7 +491,11 @@ STATIC mp_obj_t framebuf_blit(size_t n_args, const mp_obj_t *args) {
     if (n_args > 4) {
         key = mp_obj_get_int(args[4]);
     }
-
+    mp_obj_framebuf_t *palette = (mp_obj_framebuf_t *)MP_OBJ_NULL;
```

Comment:
The two `(mp_obj_framebuf_t *)MP_OBJ_NULL` (this line and below) should be just regular `NULL`.

`MP_OBJ_NULL` is an `mp_obj_t` encoding of `NULL`. But in this case you have a pointer (`mp_obj_framebuf_t*`), not an `mp_obj_t`, so you should be using `NULL`.

`mp_obj_cast_to_native_base` will return `MP_OBJ_NULL` (or a valid `mp_obj_t`), and then `MP_OBJ_TO_PTR` will convert that to `NULL`.

(What you had works because under the hood, mp_obj_t and pointers are actually (usually) encoded exactly the same way -- MP_OBJ_NULL is actually equal to NULL).